packages/cadl-ranch-specs/http/azure/resource-manager/resources/mockapi.ts

```diff
@@ -61,6 +61,490 @@ const validSingletonResource = {
   },
 };
 
+const validResourceGroupExtensionsResource = {
+  id: `/subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}/providers/Azure.ResourceManager.Resources/extensionsResources/extension`,
+  name: "extension",
+  type: "Azure.ResourceManager.Resources/extensionsResources",
+  properties: {
+    description: "valid",
+    provisioningState: "Succeeded",
+  },
+  systemData: {
+    createdBy: "AzureSDK",
+    createdByType: "User",
+    createdAt: new Date(),
+    lastModifiedBy: "AzureSDK",
+    lastModifiedAt: new Date(),
+    lastModifiedByType: "User",
+  },
+};
+
+const validSubscriptionExtensionsResource = {
+  id: `/subscriptions/${SUBSCRIPTION_ID_EXPECTED}/providers/Azure.ResourceManager.Resources/extensionsResources/extension`,
+  name: "extension",
+  type: "Azure.ResourceManager.Resources/extensionsResources",
+  properties: {
+    description: "valid",
+    provisioningState: "Succeeded",
+  },
+  systemData: {
+    createdBy: "AzureSDK",
+    createdByType: "User",
+    createdAt: new Date(),
+    lastModifiedBy: "AzureSDK",
+    lastModifiedAt: new Date(),
+    lastModifiedByType: "User",
+  },
+};
+
+const validTenantExtensionsResource = {
+  id: `/providers/Azure.ResourceManager.Resources/extensionsResources/extension`,
+  name: "extension",
+  type: "Azure.ResourceManager.Resources/extensionsResources",
+  properties: {
+    description: "valid",
+    provisioningState: "Succeeded",
+  },
+  systemData: {
+    createdBy: "AzureSDK",
+    createdByType: "User",
+    createdAt: new Date(),
+    lastModifiedBy: "AzureSDK",
+    lastModifiedAt: new Date(),
+    lastModifiedByType: "User",
+  },
+};
+
+const validResourceExtensionsResource = {
+  id: `/providers/Azure.ResourceManager.Resources/topLevelTrackedResources/top/providers/Azure.ResourceManager.Resources/extensionsResources/extension`,
```

Comment:
id is wrong. see https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/azure/resource-manager/resources/mockapi.ts#L9